src/sparse/KokkosSparse_BlockCrsMatrix.hpp

```diff
@@ -0,0 +1,940 @@
+
+    const size_type start = graph.row_map(i); // total num blocks prior to this block-row
+    const ordinal_type count = static_cast<ordinal_type> (graph.row_map(i+1) - start); // num blocks in this row
+
+    if (count == 0) {
+      return SparseBlockRowView<BlockCrsMatrix> (NULL, NULL, 1, 0);
+    } else {
+      return SparseBlockRowView<BlockCrsMatrix> (values, graph.entries, blockDim(), count, start);
+    }
+  }
+
+  /// \brief Return a SparseBlockRowViewConst of block-row i of the matrix.
+  ///
+  /// If row i does not belong to the matrix, return an empty view.
+  ///
+  /// The returned object \c view implements the following interface:
+  /// <ul>
+  /// <li> \c view.length is the number of entries (i.e. blocks) 
+  ///      in the block row </li>
+  /// <li> \c view.local_row_in_block_row(K, i) returns a nonconst pointer
+  ///      to the values of the ith local row in the k-th block of the block-row </li>
+  /// <li> \c view.full_row_in_block_row(i) returns a nonconst pointer
+  ///      to the values of the ith local row of the block-row </li>
+  /// <li> \c view.local_block_value(K, i, j) returns a nonconst reference
+  ///      to the value in the ith local row and jth local col 
+  ///      of the k-th block of the block-row </li>
+  /// <li> \c view.block(K) returns an unmanaged 2D strided Kokkos::View
+  ///      of the values of the k-th block of the block-row </li>
+  /// </ul>
+  ///
+  /// Users should not rely on the return type of this method.  They
+  /// should instead assign to 'auto'.
+  ///
+  KOKKOS_INLINE_FUNCTION
+  SparseBlockRowViewConst<BlockCrsMatrix> block_row_Const (const ordinal_type i) const {
+
+    const size_type start = graph.row_map(i); // total num blocks prior to this block-row
+    const ordinal_type count = static_cast<ordinal_type> (graph.row_map(i+1) - start); // num blocks in this row
+
+    if (count == 0) {
+      return SparseBlockRowViewConst<BlockCrsMatrix> (NULL, NULL, 1, 0);
+    } else {
+      return SparseBlockRowViewConst<BlockCrsMatrix> (values, graph.entries, blockDim(), count, start);
+    }
+  }
+
+private:
+  ordinal_type numCols_;
+  ordinal_type blockDim_; // TODO Assuming square blocks for now - add blockRowDim, blockColDim
+};
+
+//----------------------------------------------------------------------------
+//----------------------------------------------------------------------------
+
+// Assumes rows is the row_map pointer for the BlockCrsMatrix graph (i.e. cum sum of number of blocks)
+//         cols is the entries pointer for the BlockCrsMatrix graph (colidx for block-row blocks)
+//         annz is the total number of non-zeros in the CrsMatrix (blockDim*blockDim*numBlocks)
+template< typename ScalarType , typename OrdinalType, class Device, class MemoryTraits, typename SizeType >
+void
+BlockCrsMatrix<ScalarType , OrdinalType, Device, MemoryTraits, SizeType >::
+import (const std::string &label,
```

Comment:
See above comments on the name of this method.

I don't actually think this method should exist; it should at least be a hidden implementation detail.  Users don't need this.

Reply:
I don't see any usage or testing of this method for `CrsMatrix` in KokkosKernels
@crtrott is it safe for me to remove this from `CrsMatrix` and `BlockCrsMatrix`?